libqtile/backend/x11/core.py

```diff
@@ -952,3 +987,36 @@ def check_stacking(self, win: base.Window) -> None:
             self.last_focused.change_layer()
 
         self.last_focused = win
+
+    def setup_barriers(self, screens: list[Screen]) -> None:
+        if not (hasattr(self.conn, "xinputextension") and hasattr(self.conn, "xfixes")):
+            logger.warning("Unable to set pointer barriers. Missing extensions.")
+            return
+        self.conn.xinputextension.select_events(self._root.wid)
+
+        if len(screens) < 2:
+            return
+
+        borders = set()
+
+        for screen in screens:
+            for other in [s for s in screens if s is not screen]:
+                border = screen.get_joining_edge(screen, other)
```

Comment:
I'm one of those who think that static methods are a code smell... Do you mind updating `get_joining_edge` to a real method? :)

```suggestion
                border = screen.get_joining_edge(other)
```

Reply:
There was a reason I did this but I don't remember now... It does feel like an ordinary method should work!